### PR TITLE
Changes to allow for variable `ignore_index` in the cross entropy loss

### PR DESCRIPTION
**Pull request by aleximmer**

### CODE REVIEW

**Comment by wiseodd** [comment]
This is because [PyTorch and Huggingface defaults to `-100`](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html) but ASDL defaults to `-1`.

**Review by kazukiosawa — APPROVED** [review]
LGTM!
